service/src/main/kotlin/no/nav/su/se/bakover/service/søknad/SøknadServiceImpl.kt

```diff
@@ -3,18 +3,57 @@ package no.nav.su.se.bakover.service.søknad
 import arrow.core.Either
 import arrow.core.left
 import arrow.core.right
+import no.nav.su.se.bakover.database.behandling.BehandlingRepo
 import no.nav.su.se.bakover.database.søknad.SøknadRepo
+import no.nav.su.se.bakover.domain.Saksbehandler
 import no.nav.su.se.bakover.domain.Søknad
+import no.nav.su.se.bakover.service.brev.BrevService
+import org.slf4j.LoggerFactory
 import java.util.UUID
 
 internal class SøknadServiceImpl(
-    private val søknadRepo: SøknadRepo
+    private val søknadRepo: SøknadRepo,
+    private val behandlingRepo: BehandlingRepo,
+    private val brevService: BrevService
 ) : SøknadService {
+    val log = LoggerFactory.getLogger(this::class.java)
+
     override fun opprettSøknad(sakId: UUID, søknad: Søknad): Søknad {
         return søknadRepo.opprettSøknad(sakId, søknad)
     }
 
     override fun hentSøknad(søknadId: UUID): Either<FantIkkeSøknad, Søknad> {
         return søknadRepo.hentSøknad(søknadId)?.right() ?: FantIkkeSøknad.left()
     }
+
+    override fun trekkSøknad(
+        søknadId: UUID,
+        saksbehandler: Saksbehandler
+    ): Either<TrekkSøknadFeil, SøknadTrukketOk> {
+        val loggtema = "Avslutting av søknadsbehandling"
+
+        val søknad = søknadRepo.hentSøknad(søknadId)
```

Comment:
```
 val søknad = søknadRepo.hentSøknad(søknadId).getOrHandle{
            log.error("$loggtema: Fant ikke søknad")
            return TrekkSøknadFeil.FantIkkeSøknad.left()
        }
```

Reply:
getOrElse holder vel når jeg tenker meg om

Reply:
''return' is not allowed here'